typings/index.d.ts

```diff
@@ -182,6 +183,18 @@ export abstract class AnonymousGuild extends BaseGuild {
   public splashURL(options?: StaticImageURLOptions): string | null;
 }
 
+export class APIRequest {
+  private constructor(rest: unknown, method: string, path: string, options: unknown);
+  public client: Client;
+  public method: string;
+  public options: unknown;
+  public path: string;
+  private rest: unknown;
+  public retries: number;
+  public route: string;
+  private make(): Promise<Response>;
```

Comment:
Could you place private methods and properties at the top and separate them with a new line like in other classes?

Reply:
ngl, I'm fully anticipating making this an interface per point 2, just wanted to gather thoughts first, so I didn't bother formatting it yet.

Reply:
Personally I think this is fine as it is atm but sure, let's wait for more feedback on that